src/components/StudyPane/InfoPane/Motif/RootBlock.tsx

```diff
@@ -1,62 +1,64 @@
 }: {
-  id: number,
-  count: number,
-  descendants: HebWord[]
+    id: number,
+    count: number,
+    descendants: HebWord[]
 }) => {
 
-  const { ctxRootsColorMap, ctxColorAction, ctxSelectedColor, ctxSelectedHebWords, ctxSetNumSelectedWords, ctxSetSelectedHebWords } = useContext(FormatContext)
-  
-  let defaultColorFill = descendants[0].colorFill || DEFAULT_COLOR_FILL;
-  let defaultBorderColor = descendants[0].borderColor || DEFAULT_BORDER_COLOR;
-  let defaultTextColor = descendants[0].textColor || DEFAULT_TEXT_COLOR;
+  const { ctxStudyId, ctxRootsColorMap, ctxColorAction, ctxSelectedColor, ctxSelectedHebWords,
+    ctxSetNumSelectedWords, ctxSetSelectedHebWords, ctxSetRootsColorMap, ctxInViewMode } = useContext(FormatContext)
+
+  const [colorFillLocal, setColorFillLocal] = useState(DEFAULT_COLOR_FILL);
+  const [borderColorLocal, setBorderColorLocal] = useState(DEFAULT_BORDER_COLOR);
+  const [textColorLocal, setTextColorLocal] = useState(DEFAULT_TEXT_COLOR);
+  const [selected, setSelected] = useState(false);
+  const BLACK = '#000000'
 
-  const rootBlockColor = ctxRootsColorMap.get(descendants[0].strongNumber);
-  if (rootBlockColor) {
-    defaultColorFill = rootBlockColor.colorFill;
-    defaultTextColor = rootBlockColor.textColor;
+  const matchFillColor = () => {
+    let match = descendants.every((dsd) => {
+      return !dsd.colorFill || dsd.colorFill === descendants[0].colorFill;
+    });
+    return match;
   }
 
-  descendants.forEach((dsd) => {
-    if (dsd.colorFill !== defaultColorFill) { defaultColorFill = DEFAULT_COLOR_FILL; }
-    if (dsd.borderColor !== defaultBorderColor) { defaultBorderColor = DEFAULT_BORDER_COLOR; }
-    if (dsd.textColor !== defaultTextColor) { defaultTextColor = DEFAULT_TEXT_COLOR; }
-  });
+  const matchTextColor = () => {
+    let match = descendants.every((dsd) => {
+      return !dsd.textColor || dsd.textColor === descendants[0].textColor;
+    });
+    return match;
+  }
 
-  const [colorFillLocal, setColorFillLocal] = useState(defaultColorFill);
-  const [borderColorLocal, setBorderColorLocal] = useState(defaultBorderColor);
-  const [textColorLocal, setTextColorLocal] = useState(defaultTextColor);
-  const [selected, setSelected] = useState(false);
+  const matchBorderColor = () => {
+    let match = descendants.every((dsd) => {
+      return !dsd.borderColor || dsd.borderColor === descendants[0].borderColor;
+    });
+    return match;
+  }
 
   useEffect(() => {
     const rootBlockColor = ctxRootsColorMap.get(descendants[0].strongNumber);
-    if (rootBlockColor) {
-      setColorFillLocal(rootBlockColor.colorFill);
-      setTextColorLocal(rootBlockColor.textColor);
+    const matchedFillColor = matchFillColor();
+
+    if (rootBlockColor || matchedFillColor) {
+      setColorFillLocal(rootBlockColor? rootBlockColor.colorFill: descendants[0].colorFill);
+      setBorderColorLocal(rootBlockColor? rootBlockColor.borderColor!==''?rootBlockColor.borderColor:DEFAULT_BORDER_COLOR:BLACK);
```

Comment:
Why do we use the black if root block color is null? Shouldn't we use the default border color?